pkg/proxy/proxy.go

```diff
@@ -99,17 +123,42 @@ func (p *Proxy) Run(stopCh <-chan struct{}) (<-chan struct{}, error) {
 	proxyHandler.Transport = p
 	proxyHandler.ErrorHandler = p.Error
 
-	// wait for oidc auther to become ready
-	klog.Infof("waiting for oidc provider to become ready...")
-	time.Sleep(10 * time.Second)
+	// probe for readiness
+	go func() {
+		ticker := time.NewTicker(500 * time.Millisecond)
+		for {
```

Comment:
I think we can also just perform this _when_ we receive the health check.
Once we have received a health check that has passed we can always return true